src/Infrastructure/RabbitMQQueueAdapter.php

```diff
@@ -0,0 +1,129 @@
+<?php namespace StraTDeS\SharedKernel\Infrastructure;
+
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Connection\AMQPStreamConnection;
+use PhpAmqpLib\Exception\AMQPRuntimeException;
+use PhpAmqpLib\Exception\AMQPTimeoutException;
+use PhpAmqpLib\Message\AMQPMessage;
+
+class RabbitMQQueueAdapter implements QueueAdapterInterface
+{
+    /** @var RabbitMQConfigProviderInterface */
+    private $rabbitMQConfigProvider;
+
+    /** @var AMQPStreamConnection */
+    private $connection;
+
+    /** @var AMQPChannel */
+    private $channel;
+
+    public function __construct(RabbitMQConfigProviderInterface $rabbitMQConfigProvider)
+    {
+        $this->rabbitMQConfigProvider = $rabbitMQConfigProvider;
+    }
+
+    public function get(string $queueName): QueueMessage
+    {
+        $channel = $this->getChannel();
+
+        $this->declareQueue($queueName);
+
+        /** @var AMQPMessage $message */
+        $message = $channel->basic_get($queueName, false, null);
+
+        return new QueueMessage(
+            $message->delivery_info['delivery_tag'],
+            $message->getBody()
+        );
+    }
+
+    public function consume(string $queueName, callable $consumer, bool $ack = true): void
+    {
+        $channel = $this->getChannel();
+
+        $this->declareQueue($queueName);
+
+        // Set the consumer as ready to consume
+        $channel->basic_consume($queueName, '', false, !$ack, false, false, $consumer);
+
+        // Wait for callbacks of the channel (new messages)
+        while (count($channel->callbacks)) {
+            try {
+                $channel->wait();
+            } catch (AMQPTimeoutException $e) {
+                die('timeout');
```

Comment:
This shouldn't be dies, this should be our Adapter returning Infrastructure exceptions with a proper message.